src/wellKnownDidConfiguration/index.ts

```diff
@@ -61,155 +51,269 @@ function checkOrigin(input: string) {
+  {
+    expirationDate = new Date(Date.now() + 1000 * 60 * 60 * 24 * 365 * 5),
+  }: {
+    expirationDate?: Date
+  } = {}
 ): Promise<DidConfigResource> {
-  if (!Credential.isPresentation(credential)) {
-    throw new Error('Input must be an IPresentation')
-  }
-  const claimContents = credential.claim.contents
-
-  CType.verifyClaimAgainstSchema(claimContents, ctypeDomainLinkage)
-
-  const { origin } = claimContents
   checkOrigin(origin as string)
 
-  if (!(credential.claim.owner && origin)) {
-    throw new Error('Claim must have an owner and an origin property')
-  }
-  const propsToRemove = Object.keys(claimContents).filter((i) => i !== 'origin')
-  const originOnlyCredential = Credential.removeClaimProperties(credential, propsToRemove)
-
-  const {
-    proof: allProofs,
-    credentialSubject,
-    id: _,
-    legitimationIds: __,
-    ...VC
-  } = fromCredentialAndAttestation(originOnlyCredential, {
-    owner: credential.claim.owner,
-  } as any)
-
-  const ssProof = (allProofs as Proof[]).find(({ type }) => type === KILT_SELF_SIGNED_PROOF_TYPE) as SelfSignedProof
-  const digProof = (allProofs as Proof[]).find(
-    ({ type }) => type === KILT_CREDENTIAL_DIGEST_PROOF_TYPE
-  ) as CredentialDigestProof
-
-  const proof: DomainLinkageProof = {
-    ...ssProof,
-    ...digProof,
-    rootHash: credential.rootHash,
-    type: [KILT_SELF_SIGNED_PROOF_TYPE, KILT_CREDENTIAL_DIGEST_PROOF_TYPE],
-  }
-  return {
-    '@context': DID_CONFIGURATION_CONTEXT,
-    linked_dids: [
-      {
-        ...VC,
-        '@context': [DEFAULT_VERIFIABLECREDENTIAL_CONTEXT, DID_CONFIGURATION_CONTEXT],
-        expirationDate,
-        type: [DEFAULT_VERIFIABLECREDENTIAL_TYPE, DOMAIN_LINKAGE_CREDENTIAL_TYPE],
-        proof,
-        credentialSubject: {
-          id: credentialSubject['@id'] as DidUri, // canonicalize @id to id
-          origin: credentialSubject.origin as string,
-          // @ts-expect-error for compatibility with older implementations, add the credential rootHash (which is also contained in the credential id)
-          rootHash: credential.rootHash,
-        },
-      },
-    ],
+  const { didDocument } = await Did.resolveCompliant(did)
+
+  if (!didDocument) {
+    throw new Error('No Did found: Please create a Full DID')
   }
+
+  const credentials = await Promise.all([
+    createCredential(signer, origin, didDocument as ConformingDidDocument, {
+      proofType: DATA_INTEGRITY_PROOF_TYPE,
+      expirationDate,
+    }),
+    createCredential(signer, origin, didDocument as ConformingDidDocument, {
+      proofType: KILT_SELF_SIGNED_PROOF_TYPE,
+      expirationDate,
+    }),
+  ])
+
+  return didConfigResourceFromCredentials(credentials)
 }
 
-async function verifyDomainLinkageCredential(
+export async function verifyDomainLinkageCredential(
   credential: DomainLinkageCredential,
   expectedOrigin: string,
-  expectedDid?: DidUri
+  { expectedDid, allowUnsafe = false }: { expectedDid?: DidUri; allowUnsafe?: boolean } = {}
```

Comment:
`allowUnsafe` is a flag that switches whether verification allows any data to be signed, supporting the credentials that are currently being used by socialKYC and the like, or whether it requires the JSON-stringified credential to be signed, securing the credential contents against manipulation.

Reply:
Sounds good. Do we need to document this? Considering the naming people may consider it, well frankly, unsafe.